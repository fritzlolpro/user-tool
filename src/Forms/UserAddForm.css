.add-user-wrapper {
  display: inline-block;
  position: relative;
  height: 100%;
}
.add-controls {
  position: absolute !important;
  bottom: 0 !important;
  left: 0 !important;
  right: 0 !important;
  margin: auto !important;
}
.info-text {
  font-family: Roboto, sans-serif !important;
  font-weight: 400 !important;
  display: inline-block !important;
  font-size: 16px !important;
  color: #616161 !important;
  margin-bottom: 16px !important;
}
.input-fields {
  margin-right: 24px !important;
  margin-bottom: 16px !important;
}
