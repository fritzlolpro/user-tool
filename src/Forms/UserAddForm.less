.add-user-wrapper {
    display: inline-block;
    position: relative;
    height: 100%;
}

.input-fields() {
    margin-right: 24px;
    margin-bottom: 16px;
}

.add-controls() {
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    margin: auto;
}

.info-text() {
    font-family: Roboto, sans-serif;
    font-weight: 400;
    display: inline-block;
    font-size: 16px;
    color: #616161;
    margin-bottom: 16px;
}

.add-controls {
    .add-controls() !important
}

.info-text {
    .info-text() !important
}

.input-fields {
    .input-fields()  !important;
}

