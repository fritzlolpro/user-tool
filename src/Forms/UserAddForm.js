import React from 'react';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/FlatButton';
import Paper from 'material-ui/Paper';
import './UserAddForm.css';
import styles from '../Primary/UserList.css';


const UserAddForm = ({post, cancel, onChange, disablePostButton}) => {
    const onHandlePost = () => post()
    const onHandleChange = (e) => onChange(e)
    const onHandleCancel = () => cancel()
    const disable = disablePostButton()
    return (
        <Paper className='standart-card' zDepth={4}>
        <div className='add-user-wrapper'>
            <p className='info-text'>Add new user here
                <br/>
                All fields are reqiured</p>
            <div>
                <TextField
                    className='input-fields'
                    onChange={onHandleChange}
                    name='name.first'
                    floatingLabelText='User first name'/>
                <br/>
                <TextField
                    className='input-fields'
                    onChange={onHandleChange}
                    name='name.last'
                    floatingLabelText='User last name'/>
                <br/>
                <TextField
                    className='input-fields'
                    onChange={onHandleChange}
                    name='picture'
                    floatingLabelText='Url to user picture'/>
            </div>
            <div className='add-controls'>
                <RaisedButton
                    label="Post User"
                    disabled={disable}
                    primary={true}
                    onClick={onHandlePost}/>

                <FlatButton label="Cancel" onClick={onHandleCancel}/>
            </div>
            </div>
        </Paper>
    )
}

export default UserAddForm