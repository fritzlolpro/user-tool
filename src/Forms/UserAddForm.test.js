import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import UserAddForm from './UserAddForm';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

describe('UserAddForm', () => {

    const props = {
       disablePostButton: () =>{ return false }
    }

    it('should be defined', () => {
        expect(UserAddForm).toBeDefined();
    });

    it('renders without crashing'),
    () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <MuiThemeProvider>
            <UserAddForm {...props}/>
        </MuiThemeProvider>, div);
    }

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <MuiThemeProvider>
                <UserAddForm {...props}/>
            </MuiThemeProvider>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})