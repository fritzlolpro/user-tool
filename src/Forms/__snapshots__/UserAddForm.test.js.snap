// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAddForm has a valid snapshot 1`] = `
<div
  className="standart-card"
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 14px 45px rgba(0, 0, 0, 0.25),
             0 10px 18px rgba(0, 0, 0, 0.22)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
    }
  }
>
  <div
    className="add-user-wrapper"
  >
    <p
      className="info-text"
    >
      Add new user here
      <br />
      All fields are reqiured
    </p>
    <div>
      <div
        className="input-fields"
        style={
          Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": 256,
          }
        }
      >
        <label
          className={undefined}
          htmlFor="namefirst-undefined-Userfirstname-15449"
          onClick={undefined}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.3)",
              "lineHeight": "22px",
              "muiPrepared": true,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 38,
              "transform": "scale(1) translate(0, 0)",
              "transformOrigin": "left top",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "zIndex": 1,
            }
          }
        >
          User first name
        </label>
        <input
          disabled={false}
          id="namefirst-undefined-Userfirstname-15449"
          name="name.first"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "marginTop": 14,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="text"
        />
        <div>
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderColor": "#e0e0e0",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 2,
                "borderColor": "#00bcd4",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
      <br />
      <div
        className="input-fields"
        style={
          Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": 256,
          }
        }
      >
        <label
          className={undefined}
          htmlFor="namelast-undefined-Userlastname-56434"
          onClick={undefined}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.3)",
              "lineHeight": "22px",
              "muiPrepared": true,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 38,
              "transform": "scale(1) translate(0, 0)",
              "transformOrigin": "left top",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "zIndex": 1,
            }
          }
        >
          User last name
        </label>
        <input
          disabled={false}
          id="namelast-undefined-Userlastname-56434"
          name="name.last"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "marginTop": 14,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="text"
        />
        <div>
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderColor": "#e0e0e0",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 2,
                "borderColor": "#00bcd4",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
      <br />
      <div
        className="input-fields"
        style={
          Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": 256,
          }
        }
      >
        <label
          className={undefined}
          htmlFor="picture-undefined-Urltouserpicture-35735"
          onClick={undefined}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.3)",
              "lineHeight": "22px",
              "muiPrepared": true,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 38,
              "transform": "scale(1) translate(0, 0)",
              "transformOrigin": "left top",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "zIndex": 1,
            }
          }
        >
          Url to user picture
        </label>
        <input
          disabled={false}
          id="picture-undefined-Urltouserpicture-35735"
          name="picture"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "marginTop": 14,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="text"
        />
        <div>
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderColor": "#e0e0e0",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 2,
                "borderColor": "#00bcd4",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="add-controls"
    >
      <div
        className={undefined}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                   0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "minWidth": 88,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#00bcd4",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": "100%",
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "backgroundColor": false,
                  "borderRadius": 2,
                  "height": 36,
                  "muiPrepared": true,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              <span
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 14,
                    "fontWeight": 500,
                    "letterSpacing": 0,
                    "margin": 0,
                    "muiPrepared": true,
                    "opacity": 1,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "position": "relative",
                    "textTransform": "uppercase",
                    "userSelect": "none",
                  }
                }
              >
                Post User
              </span>
            </div>
          </div>
        </button>
      </div>
      <button
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            Cancel
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
